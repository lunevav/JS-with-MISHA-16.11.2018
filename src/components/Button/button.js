import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Button extends Component {
  constructor() {
    super();
    this.state = {
      value: ''
    }
  }

  _onClickHandler = () => {
      const { onClick, id } = this.props;
      // TODO ??
      const { value } = this.state;
      this.setState({ value: id });
      onClick(id);
  };

  render() {
    const {
      innerText,
      className,
      html,
      labelText,
      children,
      id
    } = this.props;
    return (
      <div>
          <div className="children">
            {html && children}
          </div>
          {labelText && <label>{labelText}</label>}
          <button
            onClick={this._onClickHandler}
            id={id}
            className={className}>
            {innerText}
            </button>
      </div>
    );
  }
}

Button.propTypes = {
  innerText: PropTypes.string,
  className: PropTypes.string,
  html: PropTypes.bool,
  labelText: PropTypes.string,
  id: PropTypes.string,
  onClick: PropTypes.func
};

Button.defaultProps = {
  id: '',
  innerText: '',
  className: '',
  html: false,
  labelText: '',
  onClick: () => {}
};

export default Button;
