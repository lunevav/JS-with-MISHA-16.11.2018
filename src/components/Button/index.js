import Button from './button';
export default Button;

// ES4-5 example
// const Button = require('./button');
// module.exports = Button;