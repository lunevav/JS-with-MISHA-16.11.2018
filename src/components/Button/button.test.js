import React from 'react';
import { mount, configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from './button';
configure({ adapter: new Adapter() });

  describe('Index', () => {
    it('should render without throwing an error', function () {
      const wrap = mount(<Button innerText="my text"/>)
      expect(wrap.find('button').text()).toBe('my text');
    })

    it('should render with label', function () {
      const component = shallow(<Button labelText="test" innerText="my text"/>)
      expect(component).toMatchSnapshot();
    })

    it('should return value on click', () => {
      const component = mount(<Button id="nashe id" innerText="my text"/>)
      const button = component.find('button');
      expect(component.state().value).toEqual('');
      button.simulate('click');
      expect(component.state().value).toEqual('nashe id');
    });
  })