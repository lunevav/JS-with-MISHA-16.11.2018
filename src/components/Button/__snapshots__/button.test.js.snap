// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    className=""
    html={false}
    id=""
    innerText="my text"
    labelText=""
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="children"
        />,
        "",
        <button
          className=""
          id=""
          onClick={[Function]}
        >
          my text
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "children",
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "my text",
          "className": "",
          "id": "",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "my text",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="children"
          />,
          "",
          <button
            className=""
            id=""
            onClick={[Function]}
          >
            my text
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "children",
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "my text",
            "className": "",
            "id": "",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "my text",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Index should render with label 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    className=""
    html={false}
    id=""
    innerText="my text"
    labelText="test"
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="children"
        />,
        <label>
          test
        </label>,
        <button
          className=""
          id=""
          onClick={[Function]}
        >
          my text
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "children",
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "my text",
          "className": "",
          "id": "",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "my text",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="children"
          />,
          <label>
            test
          </label>,
          <button
            className=""
            id=""
            onClick={[Function]}
          >
            my text
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "children",
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "my text",
            "className": "",
            "id": "",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "my text",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
